### ISSUE TITLE
AtenIntBoolOp has no lowering

### ISSUE THREAD

**Issue by renxida** [llvm/torch-mlir#3060]
AtenIntBoolOp is commonly used in onnx ops to deal with negative dims in combination with AtenMulIntOp like so:

```cpp
          Value dim = rewriter.create<Torch::AtenItemOp>(
              binder.getLoc(), rewriter.getType<Torch::IntType>(), extract);
          // deal with neg axis: if (axis < 0) axis += rank
          Value isNegative =
              rewriter.create<Torch::AtenLtIntOp>(binder.getLoc(), dim, zero);
          isNegative = rewriter.create<Torch::AtenIntBoolOp>(binder.getLoc(),
                                                             isNegative);
          Value finalOffset = rewriter.create<Torch::AtenMulIntOp>(
              binder.getLoc(), isNegative, adjustment);
          Value finalDim = rewriter.create<Torch::AtenAddIntOp>(
              binder.getLoc(), dim, finalOffset);
```

I'm getting the error:
```

./lstm.torch.mlir:68:13: error: failed to legalize operation 'torch.aten.Int.bool'
      %32 = torch.aten.select.int %arg0, %int0_22, %arg4 : !torch.vtensor<[15,2,4],f32>, !torch.int, !torch.int -> !torch.vtensor<[2,4],f32>
            ^
```

Due to there being no lowering for AtenIntBoolOp.

This error has not been surfacing in some of our test cases because AtenIntBoolOp has a folder:

```cpp
OpFoldResult AtenIntBoolOp::fold(FoldAdaptor adaptor) {
  bool b;
  if (matchPattern(getOperand(), m_TorchConstantBool(&b))) {
    return getI64IntegerAttr(getContext(), static_cast<long>(b));
  }
  return nullptr;
}
```

**Comment by renxida** [comment]
But this wouldn't work for the many many times AtenIntBoolOp is being used to deal with negative dims in TorchOnnxToTorch

![image](https://github.com/llvm/torch-mlir/assets/10362952/3ee2d398-8c0b-4413-b65b-0a2b27b1675d)

**Comment by renxida** [comment]
Ideas for how to implement AtenIntBoolOp:
- decompose to an If op that yields the two possible values
- decompose to an AtenWhereOp
- somehow grab the value of the boolean. not clear how to do that.

**Comment by renxida** [comment]
Also, maybe we should make a helper function to handle negative dims and update everything to use a different op instead of an AtenIntBool and a matmul

**Comment by renxida** [comment]
Will be fixed by #3055
